### DISCUSSION TITLE
Can anyone tell me how the lodash website generates documentation from the annotations of analysis functions？

### DISCUSSION THREAD

**Discussion by spicylemonhaha** [lodash/lodash#5811] [Q&A]
I don't see the code to generate documentation in the source code, can anyone help me？
I am building a utils library and want to automatically generate documentation by analyzing comments。

**Comment by sercanvr** [comment]
> I don't see the code to generate documentation in the source code, can anyone help me？ I am building a utils library and want to automatically generate documentation by analyzing comments。

Hello [spicylemonhaha](https://github.com/spicylemonhaha),

I just discovered your message. That's a great question, and I can see why it's confusing since the documentation generator isn't part of the core library.

From what I understand, _**Lodash**_ uses a tool called **_docdown_** to generate its documentation. docdown reads the JSDoc comments within the source code files and then creates the documentation from that data. It's likely run as a separate build step, which is why you won't find the generator's code directly within the main source.

You might want to look into docdown as a potential tool for your own utils library. It's a very effective way to generate documentation automatically by analyzing your code's comments.

> Hope this helps!
